### PR TITLE
[BEAM-7776] Stop Using Perfkit in IOIT

### PR DESCRIPTION
**Pull request by lgajowy**
I replaced Perfkit's ability to setup/teardown kubernetes with a simple bash script. Thanks to this: 

 - the tests will not be dependant on an external dependency (perfkit) and complexity is reduced
 - all logs are visible (perfkit used to hide some of them, eg from the gradle command)
 - we have more elasticity and can run multiple tests in sequence on the same k8s infrastructure
 - the bash script for kubernetes can be reused for local setup without jenkins

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status] (https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status] (https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status] (https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status] (https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run Java MongoDBIO Performance Test

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run Java MongoDBIO Performance Test

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run Java MongoDBIO Performance Test

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run Java MongoDBIO Performance Test

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run Java MongoDBIO Performance Test

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run Java MongoDBIO Performance Test

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run Java MongoDBIO Performance Test

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run Java MongoDBIO Performance Test

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run Java MongoDBIO Performance Test

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run Java MongoDBIO Performance Test

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run Java MongoDBIO Performance Test

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run Java MongoDBIO Performance Test

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run Java MongoDBIO Performance Test

**Comment by lgajowy** [comment]
@markflyhigh, @aromanenko-dev could you take a look, please?

**Comment by lgajowy** [comment]
@markflyhigh @aromanenko-dev are you available?

**Comment by lgajowy** [comment]
I just found out Alexey is on vacation - switching to @iemejia instead. :)

File: .test-infra/jenkins/Kubernetes.groovy [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  private static String namespace
+
+  private Kubernetes(job, String kubeconfigLocation, String namespace) {
+    this.job = job
+    this.kubeconfigLocation = kubeconfigLocation
+    this.namespace = namespace
+  }
+
+  /**
+   * Creates separate kubeconfig, namespace and specifies cleanup steps for job using Kubernetes.
```
[the remark this answers is not in the document]
Comment (lgajowy) [reply .test-infra/jenkins/Kubernetes.groovy]: No - I meant "separate kubeconfig, kubernetes namespace and ..." I will fix the doc

File: .test-infra/jenkins/Kubernetes.groovy [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+   */
+  static Kubernetes create(job, String kubeconfigLocation, String namespace) {
+    Kubernetes kubernetes = new Kubernetes(job, kubeconfigLocation, namespace)
+    setupKubeconfig()
+    setupNamespace()
+    addCleanupSteps()
+    return kubernetes
+  }
+
+  private static void setupNamespace() {
```
[the remark this answers is not in the document]
Comment (lgajowy) [reply .test-infra/jenkins/Kubernetes.groovy]: It's not annoying at all - I totally agree.

File: .test-infra/jenkins/job_PerformanceTests_MongoDBIO_IT.groovy [hunk 3]
```diff
@@ -15,52 +15,48 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import CommonJobProperties as common
+import Kubernetes
 
-import CommonJobProperties as commonJobProperties
-
+String kubernetesDir = '"$WORKSPACE/src/.test-infra/kubernetes"'
```
[the remark this answers is not in the document]
Comment (lgajowy) [reply .test-infra/jenkins/job_PerformanceTests_MongoDBIO_IT.groovy]: I inlined it using an existing method from commonJobProperties. is this ok?

File: .test-infra/kubernetes/kubernetes.sh [hunk 4]
```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+#    Licensed to the Apache Software Foundation (ASF) under one or more
```
[the remark this answers is not in the document]
Comment (lgajowy) [reply .test-infra/kubernetes/kubernetes.sh]: No reason. I just copied an existing header. I thought this is the convention. It looks like most of the files apply it.

File: .test-infra/kubernetes/kubernetes.sh [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+KUBECONFIG="${KUBECONFIG:=$HOME/.kube/config}"
+KUBERNETES_NAMESPACE="${KUBERNETES_NAMESPACE:=default}"
+KUBECTL="kubectl --kubeconfig=$KUBECONFIG --namespace=$KUBERNETES_NAMESPACE"
+
+function retry() {
+  local command=$1
+  local max_retries=$2
+  local sleep_time=$3
+
+  for ((i = 1; i <= $max_retries; i++)); do
```
[the remark this answers is not in the document]
Comment (lgajowy) [reply .test-infra/kubernetes/kubernetes.sh]: Did it online. posting fixes now

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run seed job

**Comment by lgajowy** [comment]
Run seed job

**Comment by aromanenko-dev** [comment]
@lgajowy Sorry for delay, I was on holidays in last 2 weeks. Since @iemejia already started a review of this PR, I'd let him finish this. In any case, I'm ready to help with this if my help will be needed.

**Comment by lgajowy** [comment]
Run Java MongoDBIO Performance Test

**Comment by lgajowy** [comment]
@aromanenko-dev sure thing - thank you for letting me know. :)

**Comment by lgajowy** [comment]
@iemejia I posted the fixes. Could you take a look again?

**Comment by lgajowy** [comment]
@iemejia pinging kindly :)

**Comment by lgajowy** [comment]
Thanks!

**Pull request merged**